grassroots-backend/src/testing/dbTransactions.ts

```diff
@@ -0,0 +1,20 @@
+import { DataSource, QueryRunner } from "typeorm";
+import { getTestApp } from "./getTestApp";
+
+let queryRunner: ReturnType<DataSource["createQueryRunner"]>;
+
+export async function startTestTransaction(): Promise<QueryRunner> {
+  const dataSource = (await getTestApp()).get(DataSource);
+  queryRunner = dataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
```

Comment:
I don't think it should block this PR. I'm wondering if we use transactions in our application, will this have an adverse effect?

Reply:
I think it would be manageable, but could get a bit clunky.
Worst case, if we're running tests, we run everything outside a transaction, which makes the tests less representative.

Postgres does have some nested transaction support, but I'm not sure how typeorm interacts with it.

Another option would be to have a "leave transactions open" param on each of our service methods that start transactions. Then the test can manually rollback the transactions when appropriate.

Happy to dig in here - thoughts on the priority?